examples/taproot-psbt.rs

```diff
@@ -0,0 +1,556 @@
+const UTXO_1: P2trUtxo = P2trUtxo {
+    txid: "9d8faa0c17dccc7c2caf52680da9c54dd061831326c639360116ae9c581f5360",
+    vout: 0,
+    script_pubkey: "51204e7980e9de99929555156d124c66624d13755eeb7a444041d7acad9cea2527f7",
+    pubkey: "d67be8c50402d16143acfe8b559d92e042c509c310b01f199d6d4932887c0a78", // Found in the "desc" field
+    master_fingerprint: "51bee341", // Prefix to derivation path in "desc" field
+    amount_in_sats: 50 * COIN_VALUE, // 50 BTC
+    derivation_path: "m/86'/1'/0'/0/0", // The derivation path for the key that can spend this UTXO (from descriptor in step 4)
+};
+
+const UTXO_2: P2trUtxo = P2trUtxo {
+    txid: "f38da1f055713b4d3eff489bf4289d66b65f691e0d1a1d5d5d6ccd7f6484a7a3",
+    vout: 0,
+    script_pubkey: "51204e7980e9de99929555156d124c66624d13755eeb7a444041d7acad9cea2527f7",
+    pubkey: "d67be8c50402d16143acfe8b559d92e042c509c310b01f199d6d4932887c0a78",
+    master_fingerprint: "51bee341",
+    amount_in_sats: 50 * COIN_VALUE,
+    derivation_path: "m/86'/1'/0'/0/0",
+};
+
+const UTXO_3: P2trUtxo = P2trUtxo {
+    txid: "921289ceed6c92ffde3d508f93376bd12a7f02f814442b9570fce8dbaa494cb2",
+    vout: 0,
+    script_pubkey: "51204e7980e9de99929555156d124c66624d13755eeb7a444041d7acad9cea2527f7",
+    pubkey: "d67be8c50402d16143acfe8b559d92e042c509c310b01f199d6d4932887c0a78",
+    master_fingerprint: "51bee341",
+    amount_in_sats: 50 * COIN_VALUE,
+    derivation_path: "m/86'/1'/0'/0/0",
+};
+
+use std::{collections::BTreeMap, str::FromStr, vec};
+
+use bitcoin::{
+    blockdata::{constants::COIN_VALUE, script::Builder as ScriptBuilder, opcodes::all::{OP_CLTV, OP_DROP, OP_CHECKSIG}},
+    consensus::{encode},
+    hashes::hex::FromHex,
+    psbt::{serialize::Serialize, Input, PartiallySignedTransaction, PsbtSighashType, Output, self},
+    schnorr::{TapTweak, self},
+    util::{
+        bip32::{DerivationPath, ExtendedPrivKey, Fingerprint, ExtendedPubKey},
+        sighash::{self, SighashCache}, taproot::{TapLeafHash, TaprootBuilder, TaprootSpendInfo, self, LeafVersion},
+    },
+    Address, Amount, OutPoint, Script, Transaction, TxIn, TxOut, Txid,
+    Witness, XOnlyPublicKey, SchnorrSighashType,
+};
+use bitcoin_hashes::Hash;
+use secp256k1::{Message, Secp256k1};
+
+type Psbt = PartiallySignedTransaction;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let secp = Secp256k1::new();
+
+    println!("\n----------------");
+    println!("\nSTART EXAMPLE 1 - P2TR with a BIP86 commitment, signed with internal key\n");
+
+    // Grab an address to receive with `bt-beneficiary getnewaddress '' bech32m`
+    let to_address = Address::from_str("bcrt1peycnn44plm8ckxaauexmf7ycmgxqs73n3yk2usnlwmmhr3e7demq295dcz")?;
+    // Get a new change address with `bt-benefactor getrawchangeaddress bech32m`
+    let change_address = Address::from_str("bcrt1pzr2yapjlxlf2qpmfjhmpy6euh5q7da8k23fl5tyhrtqe4gz0ehlsax2mp0")?;
```

Comment:
Perhaps set these in consts up the top like the utxos so there is nothing that needs editing in main?